src/librustc_codegen_ironox/type_.rs

```diff
@@ -46,25 +47,46 @@ pub enum LLType {
         args: Vec<Type>,
         ret: Type
     },
-    /// A scalar type.
+    Array {
+        length: u64,
+    },
+    PtrTo {
+        pointee: Type,
+    },
     Scalar(ScalarType),
-    /// A placeholder for types. This will be removed later.
-    None,
+    StructType {
+        name: Option<String>,
+        members: Vec<Type>
+    },
+    Void,
 }
 
 impl CodegenCx<'ll, 'tcx> {
-    crate fn type_named_struct(&self, name: &str) -> &Type {
-        unimplemented!("type_named_struct");
+    crate fn type_named_struct(&self, name: &str) -> Type {
+        let mut borrowed_types = self.types.borrow_mut();
+        let struct_type = LLType::StructType {
+            name: Some(name.to_string()),
+            members: vec![],
```

Comment:
So we start with no members, and then fill them in with the function below?

Reply:
Yep. That seems to be the way the backend-agnostic code puts together a struct.